config.go

```diff
@@ -97,11 +125,27 @@ type VaultConfig struct {
 	TLSConfig *api.TLSConfig `yaml:"tls"` // for other vault TLS options
 }
 
+// GSMConfig is the Google Secrets Manager configuration.
+type GSMConfig struct {
+	// Will add fields as needed.
+}
+
 // Mapping is a single mapping for a vault secret to a k8s secret.
 type Mapping struct {
-	// VaultPath is the path to the vault secret.
+	// SourceType is the source of a secret: Vault or GSM. Defaults to Vault.
+	SourceType string `yaml:"sourceType"`
+
+	// VaultPath is the path to a vault secret.
 	VaultPath string `yaml:"vaultPath"`
 
+	// GSMPath is the full-qualified path of a GSM secret, including its name and version.
+	// For example:
+	// - projects/*/secrets/*/versions/*
+	// - projects/*/secrets/*/versions/latest
+	// - projects/*/locations/*/secrets/*/versions/*
+	// - projects/*/locations/*/secrets/*/versions/latest
+	GSMPath string `yaml:"gsmPath"`
```

Comment:
I think this is unfortunate naming on my part.

The thing I don't think we're currently guarding against is a bad mapping where you have SourceType = gsm but the VaultPath set instead of GSMPath (and vice versa).

One possible alternatively would be to deprecate `VaultPath` and just call the new value `Path` and default the value of `Path` to whatever is in `VaultPath` in the same way you're doing with `SourceType`.  Then we could also emit a warning in the logs when you're using the old `vaultPath` key in the config.  Then there's only one `Path` and it can be shared between GSM and Vault.  What do you think?

Reply:
I'm leaning towards adding a couple of small mapping validations to cover this case. That way we can preserve the backwards compatibility and go with a `v1.5.0` release instead of changing the config API and cutting a `v2.0.0` release. I'll add that shortly.